account/manager.go

```diff
@@ -1471,14 +1550,20 @@ func (m *manager) CloseAccount(ctx context.Context, traderKey *btcec.PublicKey,
 // on-chain. These operations are performed in this order to ensure trader are
 // able to resume the spend of an account upon restarts if they happen to
 // shutdown mid-process.
-func (m *manager) spendAccount(ctx context.Context, account *Account,
+func (m *manager) spendAccount(ctx context.Context, account *Account, action Action,
 	packet *psbt.Packet, witnessType witnessType, modifiers []Modifier,
-	isClose bool, bestHeight uint32) (*Account, *wire.MsgTx, error) {
+	bestHeight uint32) (*Account, *wire.MsgTx, error) {
+
+	spendingAccount := *account
 
 	// In case we're not closing the account, let's now locate our
 	// re-created account output.
+	//
+	// TODO(ffranr): accountOutputIdx == -1 shouldn't be used to signal that
```

Comment:
It's actually only used in a single place deep down in the call stack. And the actual value is used there as well if it's not -1. So I don't think it make sense to thread through the action all the way.